engine/table/src/main/java/io/deephaven/engine/table/impl/SourcePartitionedTable.java

```diff
@@ -37,6 +38,26 @@ public class SourcePartitionedTable extends PartitionedTableImpl {
     private static final String KEY_COLUMN_NAME = "TableLocationKey";
     private static final String CONSTITUENT_COLUMN_NAME = "LocationTable";
 
+    /**
+     * Private constructor for a {@link SourcePartitionedTable}.
+     *
+     * @param table the locations table to use for this {@link SourcePartitionedTable}
+     * @param constituentDefinition The {@link TableDefinition} expected of constituent {@link Table tables}
+     * @param refreshLocations Whether the set of locations should be refreshed
+     */
+    private SourcePartitionedTable(
```

Comment:
See patch file I sent for stuff we paired on. I think we can "solve" the issue re: management transference by exposing a getting for the CSM on SourceTable, and just skip the middle man. Might result in double management by CSM, but I can live with that, I think.